src/sophys/common/devices/vpu.py

```diff
@@ -0,0 +1,35 @@
+from ophyd import Device, EpicsSignal, EpicsSignalRO, Component
+from sophys.common.utils import (
+    EpicsSignalWithMonSP,
+    EpicsSignalWithRBSP,
+    EpicsSignalMon,
+    EpicsSignalCmd,
+)
+
+
+class VPU(Device):
+    """
+    Device for controlling the VPU Undulator.
+    """
+
+    k_param = Component(EpicsSignalWithMonSP, "KParam")
```

Comment:
I'm not sure why it's called KParam, but maybe the more user-friendly name would be `gap`. At least that's how it is with IVU.

Reply:
I left it this way because the standard names of ophyd devices are based on the PV names, but we could define another pattern. I also think that gap is a better and more intuitive name.